Python/Product/ExternalProfilerDriver/VTuneStackParser.cs

```diff
@@ -0,0 +1,159 @@
+using System;
+using System.IO;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.PythonTools.Profiling.ExternalProfilerDriver
+{
+    public class PerformanceSample
+    {
+        public string Function { get; }
+        public float CPUTime { get; }
+        public string Module { get; }
+        public string FunctionFull { get; }
+        public string SourceFile { get; }
+        public string StartAddress { get; }
+
+        public PerformanceSample(string function, string cpuTime, string module, string functionFull, string sourceFile, string startAddress)
+        {
+            Function = function;
+            CPUTime = Single.Parse(cpuTime);
+            Module = module;
+            FunctionFull = functionFull;
+            SourceFile = sourceFile;
+            StartAddress = startAddress;
+        }
+    }
+
+    public class SampleWithTrace
+    {
+        private List<List<PerformanceSample>> _stacks = new List<List<PerformanceSample>>();
+        public PerformanceSample TOSFrame { get; }
+        public IEnumerable<IEnumerable<PerformanceSample>> Stacks
+        {
+            get
+            {
+                foreach (var s in _stacks)
+                {
+                    yield return s.AsEnumerable();
+                }
+            }
+        }
+
+        public SampleWithTrace(PerformanceSample sample)
+        {
+            TOSFrame = sample;
+        }
+
+        public void AddStack(List<PerformanceSample> stack)
+        {
+            _stacks.Add(stack);
+        }
+        public IEnumerable<PerformanceSample> AllSamples()
+        {
+            yield return TOSFrame; // assumes this is not null
+            foreach (var stack in Stacks)
+            {
+                foreach (var frame in stack)
+                {
+                    yield return frame;
+                }
+            }
+        }
+    }
+
+    static class VTuneStackParser
+    {
+        public static IEnumerable<string> ReadFromFile(string filePath)
+        {
+            string line;
+            using (var reader = File.OpenText(filePath))
+            {
+                while ((line = reader.ReadLine()) != null)
+                {
+                    yield return line;
+                }
+            }
+        }
+
+        public static string RemovePrePosComma(string str)
+        {
+
+            if (str.Length > 0)
+            {
+                if (str[0] == '"') { str = str.Substring(1, str.Length - 1); }
+            }
+            if (str.Length > 0)
+            {
+                if (str[str.Length - 1] == '"') { str = str.Substring(0, str.Length - 1); }
+            }
+            return str;
+        }
+
+        public static IEnumerable<SampleWithTrace> ParseFromStream(this IEnumerable<string> seq)
+        {
+            Regex startsWithComma = new Regex("^(,+)(.*)$");
```

Comment:
May as well use a regex for the entire string. That will actually simplify the rest of this function fairly dramatically. (Or possibly compare the first character with `,` and then have two separate regexes.)